Sources/Starknet/Data/Transaction/TransactionHash.swift

```diff
@@ -23,8 +26,36 @@ public class StarknetTransactionHashCalculator {
         )
     }
 
+    private class func computeCommonTransactionV3Fields(of transaction: any StarknetTransactionV3, address: Felt, chainId: StarknetChainId) -> [Felt] {
```

Comment:
function naming is off, let's make a bit better job of indicating what this does

Reply:
Imo the current name isn't that bad. It refers to [SNIP-8](https://github.com/EvyatarO/SNIPs/blob/snip-8/SNIPS/snip-8.md#protocol-changes) where the return list of this function is called `common_tx_fields`. But I agree that we could possibly find a more descriptive verb than "compute".

Reply:
We are not computing a transaction fields here, so the name is very misleading imo

Reply:
> We are not computing transaction fields here

Not sure, we are computing something called `common_tx_fields ` in the SNIP. For now the only improvement I see is `computeTransactionV3CommonTxFields`, 

We can come up with something like `computeTransactionV3CommonData`, but I would rather go for adding link to the SNIP in case this would need to refactored in the future.

Reply:
The snip says about `common_tx_fields`, it does not mention any computing. This function is also in `StarknetTransactionHashCalculator`, so someone might think that this function actually computes something connected to the transaction hash, where in fact it just compiles an ordered list of arguments that will be used to calculate the hash, right? That is why I think `prepareCommonTransactionV3Fields` or `getCommonTransactionV3Fields` may be more suitable

Reply:
> I don't mind renaming this to prepareCommonTransactionV3Fields 

Let's. 

> but we actually do generate a list in a specific order while computing the hashes (analogously to v1 method).

yup but we do not return any computed hash/address/anything like that

> Probably not as the method is private

correct me if I'm wrong, but there's no docs for swift right? so there's a chance that users will look into the code when they want to know what's up

Reply:
> but there's no docs for swift right

[There are](https://docs.swmansion.com/starknet.swift/documentation/starknet/), but I did the rename anyway